app/Listeners/RunForSavedForm.php

```diff
@@ -0,0 +1,21 @@
+<?php namespace App\Listeners;
+
+use Anomaly\SettingsModule\Setting\Form\SettingFormRepository;
+use Anomaly\Streams\Platform\Ui\Form\Event\FormWasSaved;
+
+class RunForSavedForm
+{
+
+    public function handle(FormWasSaved $event)
+    {
+        $builder = $event->getBuilder();
+
+        if (get_class($builder->getRepository()) === SettingFormRepository::class) {
+            if ($builder->getFormValues()->has('maintenance') and $builder->getFormValues()->get('maintenance')) {
+                file_put_contents(storage_path('framework/down'),
+                    json_encode(['time' => '', 'retry' => null, 'message' => null],
+                        JSON_PRETTY_PRINT));
+            }
+        }
+    }
+}
```

Comment:
one last thing @fatihalp should we put the listener in the root like this or put it in one of the modules like advs or admintheme?

Reply:
We do not use the openclassify project only for advs.Barber Projects ;)